### PR TITLE
Control the brightness of the image in the GUI.

### LINKED ISSUE #1660: brightness/contrast features for sct_label_utils -create-viewer
There are some very dark images in which it is very difficult/imposible to label.
Proposal:
- Include a slice control bar to adjust the brightness
- Include a slice bar control to adjust the contrast
- Include a check box, when active performs an automatic histogram adjustment, for example adaptive equalization, see exposure.equalize_adapthist () function in scikit-image, link below
http://scikit-image.org/docs/dev/auto_examples/color_exposure/plot_equalize.html

### PR DESCRIPTION
**Pull request by peristeri**
When right click over a image the user can control the mean and standard factor
of the image's brightness. The algorithm is the one used in the original `sct_viewer`.  Moving left / right controls the standard factor, and up / down control the mean factor.

Fixes #1660

### CODE REVIEW

**Comment by benjamindeleener** [comment]
@peristeri this works great, thank you for the addition. There is only one issue I experienced: when clicking outside the view, the viewer has a wrong behavior. You should remove all interaction outside the view.

Check the video here to see the wrong behavior: https://www.dropbox.com/s/yx1vars3yxq6hzr/sct_PR_1684.mov

File: spinalcordtoolbox/gui/base.py [hunk 1]
```diff
@@ -283,6 +285,29 @@ def __init__(self, image, params, init_values=None):
...
+        self.setup_intensity()
+
+    def setup_intensity(self):
+        if self.params.vmode == 'percentile':
+            vol = self.image.flatten()
+            vol = vol[vol > 0]
+            first_per = np.percentile(vol, self.params.vmin)
+            last_per = np.percentile(vol, self.params.vmax)
+            self.mean_intensity = np.percentile(vol, self.params.vmean)
+            self.std_intensity = last_per - first_per
```
Comment (zougloub) [inline spinalcordtoolbox/gui/base.py]: I don't think these should be called `mean/std`, it's a bit misleading.

Comment (zougloub) [reply spinalcordtoolbox/gui/base.py]: Also It doesn't look like there's a point for them to be stored in the object, they aren't used.

Comment (peristeri) [reply spinalcordtoolbox/gui/base.py]: Can you give more detail what is misleading?

**Comment by fperdigon** [comment]
@peristeri This feature will help a lot in dark images, thanks. I have been able to change the brightness and contrast, but once adjusted, if I want to modify the setting, the starting point is not the current brightness/contrast setting. The starting point depends on where the pointer is, so it is not possible to make a small adjustment with respect to the previous configuration.
A recommendation I have is to indicate that with the right click the brightness/contrast is modified, similar to how it is done with the right-left navigation

**Comment by peristeri** [comment]
@fperdigon The contrast and brightness settings are specific to that instance. To have the contrast/brightness persist for every instance is a good idea, but requires changes that are beyond the scope of this pull request.

**Comment by fperdigon** [comment]
@jcohenadad yes, that's what I was referring to

**Comment by jcohenadad** [comment]
strangely, the initial contrast/brightness is changed between the master branch. How come?

master/caf11f46d92fe8765490a53325acbf0e2b36b8f1
screen shot 2018-04-24 at 13 06 50

brightness-control/914d3ebca652911859cbaec5a004decc22c0bc36
screen shot 2018-04-24 at 13 06 18

**Comment by peristeri** [comment]
@jcohenadad In this feature, the image is initialized with a preset contrast and brightness. The initial settings are directly pulled from the sct_viewer algorithm.

**Comment by jcohenadad** [comment]
in response to [this comment](https://github.com/neuropoly/spinalcordtoolbox/pull/1684#issuecomment-384046258), why changing the original preset? the one from master works better. We don't want to "force" people to readjust contrast everytime

**Comment by peristeri** [comment]
The vmin value needed to change. In sct_viewer, it is 0.0 instead of the 5.0 used. The initial contrast should now be identical to the contrast used in sct_viewer.

**Comment by jcohenadad** [comment]
i tried it, and the initial contrast is still not the same as in master (in reference to [my comment](https://github.com/neuropoly/spinalcordtoolbox/pull/1684#issuecomment-384093614)). 

Spinal Cord Toolbox (brightness-control/11a0bbc7a6f8aa360004c56536c7fca00355c5ee)
screen shot 2018-04-25 at 23 27 33

Spinal Cord Toolbox (master/2b7b61ceae02c56fc35dd9d94acce75a73ee345e)
screen shot 2018-04-25 at 23 28 31

Please make it like it currently appears on master.

Also, the text "use the left/right key..." appears in another part of the window, separate from another message related to image navigation ("use left/right arrow...") which is kind of awkward. I would put all messages at the same location, e.g., next line.

**Pull request merged**